# Computational Tools for Host Prediction

Here are some publications and tools that describe how to predict virus host interactions or tools that could be used to do so.

# Overviews of phage host interactions

* [Phage-Host Interaction: an Ecological Perspective](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC419959/)

# Overview of Computational Predictions

The paper [Computational approaches to predict bacteriophage–host relationships](https://academic.oup.com/femsre/article/40/2/258/2570202) by Rob, Kate, Karoline, Jeroen, and Bas describes different approaches that could be used to calculate phage/host relationships. Not all work well, but most work.

A review of [An Integrative Approach to Virus–Host Protein–Protein Interactions](https://link.springer.com/protocol/10.1007/978-1-4939-8618-7_8)

[Computational prediction of inter-species relationships through omics data analysis and machine learning](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-018-2388-7) 

[Computational Prediction of Host-Pathogen Interactions Through Omics Data Analysis and Machine Learning](https://link.springer.com/chapter/10.1007/978-3-319-56154-7_33)

[Phage hunters: Computational strategies for finding phages in large-scale ‘omics datasets](https://www.sciencedirect.com/science/article/pii/S0168170217306019)

# Approaches to predict phage hosts

* [Viral host prediction with Deep Learning](https://www.biorxiv.org/content/10.1101/575571v1.abstract)
* [Identifying viruses from metagenomic data by deep learning](https://arxiv.org/abs/1806.07810)
* [PHISDetector: a web tool to detect diverse in silico phage-host interaction signals](https://www.biorxiv.org/content/10.1101/661074v1.abstract)
* [Structure-based prediction of host–pathogen protein interactions](https://www.sciencedirect.com/science/article/pii/S0959440X16301245)
* [HostPhinder: A Phage Host Prediction Tool](https://www.mdpi.com/1999-4915/8/5/116)
* [Linking Virus Genomes with Host Taxonomy](https://www.mdpi.com/1999-4915/8/3/66)
* [A network-based integrated framework for predicting virus-host interactions](https://www.biorxiv.org/content/10.1101/505768v2)
* [Prediction of virus-host infectious association by supervised learning methods](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5374558/)
* [Host Taxon Predictor - A Tool for Predicting Taxon of the Host of a Newly Discovered Virus](https://github.com/wojciech-galan/viruses_classifier) and [paper](https://www.nature.com/articles/s41598-019-39847-2)
* [VirHostMatcher](https://github.com/jessieren/VirHostMatcher) and [paper](https://academic.oup.com/nar/article/45/1/39/2605663)


# Identifying phage sequences in metagenomes

* [PhaMers identifies novel bacteriophage sequences from thermophilic hot springs](https://www.biorxiv.org/content/10.1101/169672v1.abstract)
* [MARVEL, a Tool for Prediction of Bacteriophage Sequences in Metagenomic Bins](https://www.frontiersin.org/articles/10.3389/fgene.2018.00304/full)
* [WIsH: who is the host? Predicting prokaryotic hosts from metagenomic phage contigs](https://academic.oup.com/bioinformatics/article/33/19/3113/3964377)
* [VirFinder: a novel k-mer based tool for identifying viral sequences from assembled metagenomic data](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0283-5)
* [ViraMiner: Deep Learning on Raw DNA Sequences for Identifying Viral Genomes in Human Samples](https://www.biorxiv.org/content/10.1101/602656v2)
* [VirSorter: mining viral signal from microbial genomic data](https://peerj.com/articles/985/)
* [Robust taxonomic classification of uncharted microbial sequences and bins with CAT and BAT](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1817-x)
* [MetaVir](http://metavir-meb.univ-bpclermont.fr/) and [paper](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-15-76)

# Predicting Environments
* [Metagenomic Approaches to Assess Bacteriophages in Various Environmental Niches](https://www.mdpi.com/1999-4915/9/6/127) 

# Prophage Finding

**You should start with these two classics:** [Prophages and bacterial genomics: what have we learned so far?
](https://onlinelibrary.wiley.com/doi/full/10.1046/j.1365-2958.2003.03580.x) and [Prophage Genomics](https://mmbr.asm.org/content/67/2/238)

* [PhiSpy: a novel algorithm for finding prophages in bacterial genomes that combines similarity- and composition-based strategies](https://github.com/linsalrob/PhiSpy) and [paper](https://academic.oup.com/nar/article/40/16/e126/1027055)
* [Phage_Finder: Automated identification and classification of prophage regions in complete bacterial genome sequences](https://academic.oup.com/nar/article/34/20/5839/3100473)
* [PHAST, PHASTER and PHASTEST: Tools for finding prophage in bacterial genomes](https://academic.oup.com/bib/article/20/4/1560/4222653)
    * [PHAST: A Fast Phage Search Tool](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3125810/)
    * [PHASTER: a better, faster version of the PHAST phage search tool](https://academic.oup.com/nar/article/44/W1/W16/2499364)
* [Prophage Hunter: an integrative hunting tool for active prophages](https://academic.oup.com/nar/article/47/W1/W74/5494712)
* [ProphET, prophage estimation tool: A stand-alone prophage sequence prediction tool with self-updating reference database](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0223364)
* [PhageWeb](http://computationalbiology.ufpa.br/phageweb) and [paper](https://www.frontiersin.org/articles/10.3389/fgene.2018.00644/full) *Note:* this appears to not be available any longer, although there is this [API](https://github.com/phagewebufpa/API)

# Other tools and approaches

## Classifying Sequences

* [Classifying the Unclassified: A Phage Classification Method](https://www.mdpi.com/1999-4915/11/2/195)

## Finding ends

* [PhageTerm: a tool for fast and accurate determination of phage termini and packaging mechanism using next-generation sequencing data](https://sourceforge.net/projects/phageterm/) and [paper](https://www.nature.com/articles/s41598-017-07910-5)

# Databases

* [VOGDB](http://vogdb.org/)
* [pVOG](http://dmk-brain.ecn.uiowa.edu/pVOGs/) and [paper](https://www.ncbi.nlm.nih.gov/pubmed/27789703)
* [Virus-Host DB](https://www.genome.jp/virushostdb/)

# CRISPr Finding

* [CRISPRdigger](https://www.nature.com/articles/srep32942)
* [CRISPRCasFinder](https://crisprcas.i2bc.paris-saclay.fr) 
    * [CRISPRfinder](https://crispr.i2bc.paris-saclay.fr/Server/) is the old version of CRISPRCasFinder and [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933234/)
    * [CRISPRdb](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-172) is the related spacer database
* [PILER-CR](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-18) 
* [CRISPR recognition tool (CRT)](https://www.ncbi.nlm.nih.gov/pubmed/17577412)
* [CRISPI](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btp586) is another database 
* [CRISPRDirection](https://omictools.com/crisprdirection-tool) and [paper](https://www.ncbi.nlm.nih.gov/pubmed/24578404)
* [CRISPRmap](http://rna.informatik.uni-freiburg.de/CRISPRmap/) and [paper](https://academic.oup.com/nar/article/41/17/8034/2411525)
* [CRISPRTarget](http://bioanalysis.otago.ac.nz/CRISPRTarget/)
* [CRISPRViz](https://github.com/CRISPRlab/CRISPRviz) and [paper](https://www.tandfonline.com/doi/abs/10.1080/15476286.2018.1493332?journalCode=krnb20)
* [CRISPRStudio](https://github.com/moineaulab/CRISPRStudio)
* [metaCRAST](https://github.com/molleraj/MetaCRAST) and [paper](https://peerj.com/articles/3788/)
* [metaCRISPR](https://github.com/hangelwen/metaCRISPR) and [paper](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btw456)
* [crass](https://github.com/ctSkennerton/crass) (but not the real crAss) and [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3664793/?report=reader)

