# Computational Tools for Host Prediction

Here are some publications and tools that describe how to predict virus host interactions or tools that could be used to do so.

# Overview of Computational Predictions

The paper [Computational approaches to predict bacteriophageâ€“host relationships](https://academic.oup.com/femsre/article/40/2/258/2570202) by Rob, Kate, Karoline, Jeroen, and Bas describes different approaches that could be used to calculate phage/host relationships. Not all work well, but most work.


# Approaches to predict phage hosts

* [Viral host prediction with Deep Learning](https://www.biorxiv.org/content/10.1101/575571v1.abstract)
* [PHISDetector: a web tool to detect diverse in silico phage-host interaction signals](https://www.biorxiv.org/content/10.1101/661074v1.abstract)

# Identifying phage sequences in metagenomes
* [PhaMers identifies novel bacteriophage sequences from thermophilic hot springs](https://www.biorxiv.org/content/10.1101/169672v1.abstract)

# Prophage Finding



# CRISPr Finding

* [CRISPRdigger](https://www.nature.com/articles/srep32942)
* [CRISPRCasFinder](https://crisprcas.i2bc.paris-saclay.fr) 
    * [CRISPRfinder](https://crispr.i2bc.paris-saclay.fr/Server/) is the old version of CRISPRCasFinder and [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1933234/)
    * [CRISPRdb](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-172) is the related spacer database
* [PILER-CR](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-8-18) 
* [CRISPR recognition tool (CRT)](https://www.ncbi.nlm.nih.gov/pubmed/17577412)
* [CRISPI](https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btp586) is another database 